conjur/api/api.py

```diff
@@ -410,27 +410,37 @@ def whoami(self) -> dict:
 
         return json.loads(json_response.decode('utf-8'))
 
-    def list_members_of(self, uri_parameters: ListMembersOfData = None) -> dict:
+    def list_members_of_role(self, uri_parameters: ListMembersOfData = None) -> list:
         """
         List members within a role.
         """
+        if not uri_parameters.resource or not uri_parameters.resource.identifier:
+            raise MissingRequiredParameterException("Missing required parameter, 'identifier'")
+
+        if not uri_parameters.resource or not uri_parameters.resource.kind:
+            raise MissingRequiredParameterException("Missing required parameter, 'kind'")
+
         params = {
             'account': self._account,
-            'identifier': uri_parameters.identifier,
-            'kind': uri_parameters.kind,
+            'identifier': uri_parameters.resource.identifier,
+            'kind': uri_parameters.resource.kind,
         }
         params.update(self._default_params)
 
         request_parameters = uri_parameters.list_dictify()
         del request_parameters['identifier']
+        del request_parameters['resource']
         json_response = invoke_endpoint(HttpVerb.GET,
                                         ConjurEndpoint.ROLES_MEMBERS_OF,
                                         params,
                                         query=request_parameters,
                                         api_token=self.api_token,
                                         ssl_verify=self._ssl_verify).content
 
-        return json.loads(json_response.decode('utf-8'))
+        resources = json.loads(json_response.decode('utf-8'))
+        # For each element (resource) in the resources sequence, we extract the resource id
+        resource_list = map(lambda resource: resource['member'], resources)
```

Comment:
Why this is needed?
Why not returning the response as is?

Reply:
Response contains the entire json and not just the id